packages/xo-server/src/vhd-merge.integ.spec.js

```diff
@@ -0,0 +1,203 @@
+/* eslint-env jest */
+
+import execa from 'execa'
+import fs from 'fs-extra'
+import path from 'path'
+import rimraf from 'rimraf'
+import LocalHandler from './remote-handlers/local.js'
+import { tmpDir, pFromCallback } from './utils'
+import vhdMerge, { chainVhd, Vhd } from './vhd-merge'
+
+const initialDir = process.cwd()
+
+jest.setTimeout(10000)
+
+beforeEach(async () => {
+  const dir = await tmpDir()
+  process.chdir(dir)
+})
+
+afterEach(async () => {
+  const tmpDir = process.cwd()
+  process.chdir(initialDir)
+  await pFromCallback(cb => rimraf(tmpDir, cb))
+})
+
+async function createRandomFile (name, sizeMo) {
+  await execa('bash', [
+    '-c',
+    `< /dev/urandom tr -dc "\\t\\n [:alnum:]" | head -c ${sizeMo}M >${name}`,
+  ])
+}
+
+async function checkFile (vhdName) {
+  await execa('vhd-util', ['check', '-p', '-b', '-t', '-n', vhdName])
+}
+
+async function recoverRawContent (vhdName, rawName, originalSize) {
+  await checkFile(vhdName)
+  await execa('qemu-img', ['convert', '-fvpc', '-Oraw', vhdName, rawName])
+  await execa('truncate', ['-s', originalSize, rawName])
+}
+
+async function convertFromRawToVhd (rawName, vhdName) {
+  await execa('qemu-img', ['convert', '-f', 'raw', '-Ovpc', rawName, vhdName])
+}
+
+test('writeData on empty file', async () => {
+  expect.assertions(1)
+  const moOfRandom = 11
+  await createRandomFile('randomfile', moOfRandom)
+  await execa('qemu-img', ['create', '-fvpc', 'empty.vhd', moOfRandom + 'M'])
+  const randomData = await fs.readFile('randomfile')
+  const handler = new LocalHandler({ url: 'file://' + process.cwd() })
+  const originalSize = await handler.getSize('randomfile')
```

Comment:
Why is it necessary? Isn't it the same as `moOfRandom`?

Reply:
I am always a bit suspicious with file sizes, that it could have been rounded up.